-Wclass-memaccess warnings compiling with GCC 8

**Godot version:**
Master (4b6e7b5)

**OS/device including version:**
Linux, GCC 8.

**Issue description:**
Those two warnings are raised when compiling with `scons p=x11 dev=yes` (`tools=yes target=debug`):
```
In file included from ./core/dvector.h:34,
                 from ./core/variant.h:41,
                 from ./core/method_ptrcall.h:36,
                 from ./core/method_bind.h:35,
                 from ./core/class_db.h:34,
                 from ./core/resource.h:34,
                 from ./core/io/resource_loader.h:34,
                 from modules/pvr/texture_loader_pvr.h:34,
                 from modules/pvr/texture_loader_pvr.cpp:31:
modules/pvr/texture_loader_pvr.cpp: In function 'void _compress_pvrtc4(Image*)':
./core/os/copymem.h:44:56: warning: 'void* memcpy(void*, const void*, size_t)' writing to an object of non-trivially copyable type 'class Javelin::ColorRgba<unsigned char>'; use copy-assignment or copy-initialization instead [-Wclass-memaccess]
 #define copymem(to, from, count) memcpy(to, from, count)
                                                        ^
modules/pvr/texture_loader_pvr.cpp:218:4: note: in expansion of macro 'copymem'
    copymem(bm.GetData(), &r[ofs], size);
    ^~~~~~~
In file included from thirdparty/pvrtccompressor/PvrTcEncoder.h:4,
                 from modules/pvr/texture_loader_pvr.cpp:32:
thirdparty/pvrtccompressor/ColorRgba.h:81:7: note: 'class Javelin::ColorRgba<unsigned char>' declared here
 class ColorRgba : public ColorRgb<T> {
       ^~~~~~~~~
```
```
In file included from ./core/dvector.h:34,
                 from ./core/image.h:35,
                 from ./servers/visual_server.h:34,
                 from ./servers/visual/rasterizer.h:35,
                 from servers/visual/visual_server_scene.h:34,
                 from servers/visual/visual_server_scene.cpp:31:
servers/visual/visual_server_scene.cpp: In member function 'void VisualServerScene::_update_instance_lightmap_captures(VisualServerScene::Instance*)':
./core/os/copymem.h:45:47: warning: 'void* memset(void*, int, size_t)' clearing an object of non-trivial type 'struct Color'; use assignment or value-initialization instead [-Wclass-memaccess]
 #define zeromem(to, count) memset(to, 0, count)
                                               ^
servers/visual/visual_server_scene.cpp:1243:2: note: in expansion of macro 'zeromem'
  zeromem(p_instance->lightmap_capture_data.ptrw(), 12 * sizeof(Color));
  ^~~~~~~
In file included from ./core/image.h:34,
                 from ./servers/visual_server.h:34,
                 from ./servers/visual/rasterizer.h:35,
                 from servers/visual/visual_server_scene.h:34,
                 from servers/visual/visual_server_scene.cpp:31:
./core/color.h:39:8: note: 'struct Color' declared here
 struct Color {
        ^~~~~
```

Input from @reduz:
```
11:24 <reduz> first one no idea, second one seems maybe clearing manually may be better (or not using color)
```

Same on MinGW (gcc 8.2.0) with addition of few extra warnings:
```
drivers/rtaudio/audio_driver_rtaudio.cpp: In member function 'virtual Error AudioDriverRtAudio::init()':
drivers/rtaudio/audio_driver_rtaudio.cpp:132:12: warning: catching polymorphic type 'class RtAudioError' by value [-Wcatch-value=]
   } catch (RtAudioError) {
            ^~~~~~~~~~~~
```

```
drivers/wasapi/audio_driver_wasapi.cpp: In member function 'virtual ULONG CMMNotificationClient::Release()':
drivers/wasapi/audio_driver_wasapi.cpp:93:11: warning: deleting object of polymorphic class type 'CMMNotificationClient' which has non-virtual destructor might cause undefined behavior [-Wdelete-non-virtual-dtor]
    delete this;
           ^~~~
```

--- a/modules/pvr/texture_loader_pvr.cpp
+++ b/modules/pvr/texture_loader_pvr.cpp
@@ -33,6 +33,7 @@
 #include "RgbaBitmap.h"
 #include "core/os/file_access.h"
 #include <string.h>
+#include <new>
 
 static void _pvrtc_decompress(Image *p_img);
 
@@ -215,12 +216,10 @@ static void _compress_pvrtc4(Image *p_img) {
 			int ofs, size, w, h;
 			img->get_mipmap_offset_size_and_dimensions(i, ofs, size, w, h);
 			Javelin::RgbaBitmap bm(w, h);
-			copymem(bm.GetData(), &r[ofs], size);
-			{
+			for (unsigned j = 0; j < size / 4; j++) {
 				Javelin::ColorRgba<unsigned char> *dp = bm.GetData();
-				for (int j = 0; j < size / 4; j++) {
-					SWAP(dp[j].r, dp[j].b);
-				}
+				/* red and Green colors are swapped.  */
+				new (dp) Javelin::ColorRgba<unsigned char>(r[ofs + 4 * j + 2], r[ofs + 4 * j + 1], r[ofs + 4 * j], r[ofs + 4 * j + 3]);
 			}
 
 			new_img->get_mipmap_offset_size_and_dimensions(i, ofs, size, w, h);

--- a/servers/visual/visual_server_scene.cpp
+++ b/servers/visual/visual_server_scene.cpp
@@ -32,6 +32,7 @@
 #include "core/os/os.h"
 #include "visual_server_global.h"
 #include "visual_server_raster.h"
+#include <new>
 /* CAMERA API */
 
 RID VisualServerScene::camera_create() {
@@ -1240,7 +1241,9 @@ void VisualServerScene::_update_instance_lightmap_captures(Instance *p_instance)
 
 	//print_line("update captures for pos: " + p_instance->transform.origin);
 
-	zeromem(p_instance->lightmap_capture_data.ptrw(), 12 * sizeof(Color));
+	for (int i = 0; i < 12; i++)
+		new (&p_instance->lightmap_capture_data.ptrw()[i]) Color;
+
 	//this could use some sort of blending..
 	for (List<Instance *>::Element *E = geom->lightmap_captures.front(); E; E = E->next()) {
 		const PoolVector<RasterizerStorage::LightmapCaptureOctree> *octree = VSG::storage->lightmap_capture_get_octree_ptr(E->get()->base);

